src/material/core/datetime/date-selection.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DateAdapter} from '@angular/material/core';
+import {Subject} from 'rxjs';
+
+export abstract class MatDateSelection<D> {
```

Comment:
I think it was renamed in a later commit to `MatDateSelectionModel` which better describes the intent of the class. Let's rename it now to keep this branch always in a state where we're happy with what's on it.

Reply:
Yes.  Do you want me to revert this pull request and create another one?  I already copied the commit you are talking about to my repo.

Reply:
If that is easier for you. I'm not sure what other changes might be included in that commit, I'd like to keep this one just focused on creating the selection model classes

Reply:
this pull request gets messy, so I created a new fork, addressed your commends and created a new pull request.  Please take a look.  Thanks.